### ISSUE TITLE
About soft dice loss

### ISSUE THREAD

**Issue by jeongHwarr** [qubvel/segmentation_models.pytorch#567]
Hello. Thank you for your project! 

I made a pull request with the following contents, but I was not sure and closed it.

I recently changed the code from using dice loss in `utils` to using soft dice score in `losses` folder.

https://github.com/qubvel/segmentation_models.pytorch/pull/566#issue-1146457033

Then, the performance fell and I thought there was a problem with the code, so I requested pull. But I realized that the code I requested and your code are not mathematically different.

```
def soft_dice_score(
    output: torch.Tensor,
    target: torch.Tensor,
    smooth: float = 0.0,
    eps: float = 1e-7,
    dims=None,
) -> torch.Tensor:
    assert output.size() == target.size()
    if dims is not None:
        intersection = torch.sum(output * target, dim=dims)
        cardinality = torch.sum(output + target, dim=dims)
        fp = torch.sum(output * (1.0 - target), dim=dims)
        fn = torch.sum((1 - output) * target, dim=dims)
        test = 2 * intersection + fn + fp
        test2 = cardinality - test
    else:
        intersection = torch.sum(output * target)
        cardinality = torch.sum(output + target)
        fp = torch.sum(output * (1.0 - target))
        fn = torch.sum((1 - output) * target)
        test = 2 * intersection + fn + fp
        test2 = cardinality - test
    dice_score = (2.0 * intersection + smooth)/(2 * intersection + fn + fp + smooth).clamp_min(eps)
    dice_score_2 = (2.0 * intersection + smooth) / (cardinality + smooth).clamp_min(eps)
    return dice_score_2
```

`dice_score` and `dice_score_2` are mathematically equivalent. But the result is different. It's probably a numercial issue, but I don't know why it's different. (The same seed was used for the test, and the same result is always obtained when each test is performed.)

When `dice_score` was used instead of `dice_score_2`, the test performance was good. I don't know what the difference between these two is.
